build-system/build-targets.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script sets the build targets for our PR check, where the build targets
+ * determine which tasks are required to run for pull request builds.
+ */
+const {bold, cyan, yellow} = require('ansi-colors');
+const path = require('path');
+const {gitDiffNameOnlyMaster} = require('./git');
+
+const ALL_TARGETS = ['BUNDLE_SIZE', 'OWNERS', 'PR_DEPLOY', 'TEST_STATUS'];
+
+/**
+ * Determines if a file is a filetype containing code.
+ *
+ * @param {string} filePath
+ * @return {boolean}
+ */
+function isCode(filePath) {
+  const fileName = path.basename(filePath);
+  return !(
+    fileName.startsWith('.') ||
```

Comment:
That may introduce additional work down the road because in the case that we change or add any dot file like `.env` that affects tests, someone would have to update this.

We should lean towards running tests as opposed to skipping them.

Reply:
Fair enough, I'll drop this bit and just run it for any changes to the root directory

Reply:
> We should lean towards running tests as opposed to skipping them.

This is the kicker.

+1 to getting rid of `isCode` and defaulting to running all tests for changes that don't obviously belong to one particular app (like dotfiles in the root dir).

There's one worthwhile optimization though: Don't run tests against PRs that only update documentation. (`amphtml` runs `gulp check-links`. May be worth adding here in a separate PR. There's also `gulp lint` and `gulp prettify` that check code / documentation style against local changes no matter what the PR does.)

Reply:
Just seeing this, but already removed `isCode` per recs from @danielrozenberg and @estherkim . We don't have gulp set up here, but `pr-check` runs `eslint .` before any build target checking. As for excluding documentation, I think I'll put that in as a separate PR.